cpp/include/cudf/interop.hpp

```diff
@@ -37,6 +39,14 @@ struct ArrowArray;
 
 struct ArrowArrayStream;
 
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+// These are types from arrow that we are forward declaring for our API to
+// avoid needing to include nanoarrow headers.
+typedef int32_t ArrowDeviceType;  // NOLINT
+
+#define ARROW_DEVICE_CUDA 2
+#endif
+
```

Comment:
You can use `@cond` to have doxygen ignore content
```suggestion
///@cond
// These are types from arrow that we are forward declaring for our API to
// avoid needing to include nanoarrow headers.
typedef int32_t ArrowDeviceType;  // NOLINT

#define ARROW_DEVICE_CUDA 2
///@endcond

```
Note the triple `///`. You can use `//!< @cond` and `//!< @endcond` too

Reply:
Yup I'm familiar. Sure, I can switch to that approach if you prefer.